### PR TITLE
feat: Keyboard shortcuts for component testing

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
Implement shortcuts for runner-ct:

https://user-images.githubusercontent.com/16926049/107085068-9a1fdc00-6800-11eb-9664-35898d8f113f.mov

> Using real clicks we can ensure that keyboard events from a real keyboard will work 🙌

### CODE REVIEW

**Review by elevatebart — CHANGES_REQUESTED** [review]
When using the npm/vue command `cy:open`, the overflow is all broken

Screen Shot 2021-02-05 at 3 08 40 PM

Can you check the css changes ?

File: packages/runner-ct/src/SpecList/SpecList.tsx [hunk 1]
```diff
@@ -11,16 +12,41 @@ interface SpecsListProps {
   selectedSpecs: string[]
   specs: Cypress.Cypress['spec'][]
   onSelectSpec: OnSelectSpec
-  disableTextSelection: boolean
+  disableTextSelection?: boolean
 }
 
-export const SpecList: React.FC<SpecsListProps> = (props) => {
+export const SpecList: React.FC<SpecsListProps> = observer((props) => {
```
Comment (lmiller1990) [inline packages/runner-ct/src/SpecList/SpecList.tsx:18]: what does wrapping this with `observer` do? does it fix something specifically or just best practice?

File: packages/runner-ct/src/SpecList/SpecList.tsx [hunk 2]
```diff
@@ -11,16 +12,41 @@ interface SpecsListProps {
...
+
+      if (specElement) {
+        specElement.focus()
+      }
+    }
+
+    const selectedSpecIndex = props.specs.findIndex((spec) =>
+      spec.relative === (document.activeElement as HTMLElement)?.dataset?.spec)
+
+    if (e.key === 'ArrowUp') {
```
Comment (lmiller1990) [inline packages/runner-ct/src/SpecList/SpecList.tsx:39]: nice, I like this much better than tab/shift tab

File: packages/runner-ct/src/SpecList/components/SearchSpec/SearchSpec.tsx [hunk 3]
```diff
@@ -7,9 +8,25 @@ interface SearchSpecProps {
 }
 
 export const SearchSpec: React.FC<SearchSpecProps> = (props) => {
+  const inputRef = React.useRef<HTMLInputElement>(null)
+  // const ignoreSlashInput
+
+  React.useEffect(() => {
```
Comment (lmiller1990) [inline packages/runner-ct/src/SpecList/components/SearchSpec/SearchSpec.tsx:14]: I think we should also need to open the spec list if it is not already open

**Review by lmiller1990 — COMMENTED** [review]
I think `/` needs to open the spec list if it is not already open.

Code seems fine!

File: packages/runner-ct/src/SpecList/SpecList.tsx (hunk 1 above)
Comment (dmtrKovalenko) [reply packages/runner-ct/src/SpecList/SpecList.tsx:18]: Yes, I was facing a problem of just unexpected crash inside useEffect that uses observable. It was just a crash that handled by internal react `try {} catch` without any console.log. If you interested – try to delete it and observe

That's why I asked you to add this observer 😉

File: packages/runner-ct/src/SpecList/spec-file.scss [hunk 4]
```diff
@@ -4,14 +4,18 @@
   white-space: nowrap;
   overflow-x: hidden;
   text-overflow: ellipsis;
+  outline-offset: 4px;
 }
 
 .spec-list__spec-file__radio {
   margin-bottom: 8px;
+  cursor: pointer;
 }
```
Comment (elevatebart) [inline packages/runner-ct/src/SpecList/spec-file.scss:13]:
```suggestion
  outline-offset: 4px;
}

.spec-list__spec-file__radio {
  margin-bottom: 8px;
  cursor: pointer;
  white-space: nowrap;
  overflow-x: hidden;
  text-overflow: ellipsis;
}
```

**Review by elevatebart — CHANGES_REQUESTED** [review]
The change in CSS breaks  the spec file text-overflow.

Screen Shot 2021-02-08 at 10 54 55 AM

After that it's all good.

**Review by elevatebart — DISMISSED** [review]
LGTM

**Review by lmiller1990 — DISMISSED** [review]
Nice - just needs the conflicts to be resolved.

**Review by elevatebart — APPROVED** [review]
LGTM

**Pull request merged**